sdk/core/System.ClientModel/src/Pipeline/ClientPipeline.RequestOptionsProcessor.cs

```diff
@@ -185,7 +185,7 @@ private bool TryGetCustomPerTryPolicy(int index, out PipelinePolicy policy)
 
         private bool TryGetFixedPerTransportPolicy(int index, out PipelinePolicy policy)
         {
-            if (index < _perTryIndex + _customPerCallPolicies.Length + _customPerTryPolicies.Length)
+            if (index < _beforeTransportIndex + _customPerCallPolicies.Length + _customPerTryPolicies.Length)
```

Comment:
Is this fixing a bug?  Curious why it's needed.

Reply:
Yeah I think this is a bug. [This test](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/System.ClientModel/tests/Pipeline/ClientPipelineTests.cs#L295) was failing with an index out of range exception after adding a default logging policy. The logging policy was not getting picked up by this method, TryGetFixedPerTransportPolicy. This if statement was the same as TryGetCustomPerTryPolicy so it was never evaluated as true.

Based on the diagram in [this comment](https://github.com/Azure/azure-sdk-for-net/blob/bae9e6d9b76ca084a98be2ee1b8a2aba3e1a040b/sdk/core/System.ClientModel/src/Pipeline/ClientPipeline.RequestOptionsProcessor.cs#L88), I think the fixed/custom transport policies are based on the before transport index.

Reply:
I see why Krzysztof is not wild about this type's implementation :)

I'll have a look at this and see if I can add some more-helpful comments here as well, and why existing tests didn't already catch this issue!  Thanks for finding and fixing and I'll follow up with my learnings.

Reply:
I think the tests didn't catch this because currently _perTryIndex and _beforeTransportIndex are the same because there were not any fixed per transport defined in SCM. So it is a bug but only now that there's a logging policy. I updated the test in this PR so I think that should be sufficient to cover this case.